.github/workflows/ci.yml

```
@@ -0,0 +1,56 @@
+          enable-cache: true   # enables automatic actions/cache reuse
+
+      - name: Install dev dependencies with uv
+        working-directory: api-server
+        run: |
+          uv sync --group dev
+      - name: Run tests with pytest and coverage
+        working-directory: api-server
+        run: |
+          uv run pytest --cov=app --cov-report=xml --cov-report=term-missing -v -q --junitxml=pytest-report.xml
+          - name: Upload coverage reports to Codecov
+  uses: codecov/codecov-action@v5
```

actionlint (1.7.7)

"steps" section is missing in job "uses"

(syntax-check)